rotkehlchen/history/price.py

```diff
@@ -119,11 +119,40 @@ def set_oracles_order(oracles: Sequence[HistoricalPriceOracle]) -> None:
         instance._oracles = oracles
         instance._oracle_instances = [getattr(instance, f'_{oracle!s}') for oracle in oracles]
 
+    @staticmethod
+    def _get_cached_price_or_query(
+            from_asset: Asset,
+            to_asset: Asset,
+            timestamp: Timestamp,
+            max_seconds_distance: int | None,
+    ) -> Price | None:
+        """Helper method to get price from cache if `max_seconds_distance`
+        is given, otherwise query normally.
+        """
+        if from_asset == to_asset:
+            return Price(ONE)
+
+        if max_seconds_distance is not None:
+            cached_price_entry = GlobalDBHandler.get_historical_price(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                timestamp=timestamp,
+                max_seconds_distance=max_seconds_distance,
+            )
+            return cached_price_entry.price if cached_price_entry is not None else None
```

Comment:
shouldn't it revert to a normal query instead of None if there is no cached entry?

Reply:
a normal query will make a remote call and the code path is specifically for retrieving prices cached in the globaldb.

i added a comment to mention that in the code.